src/components/trace-element.vue

```diff
@@ -1,17 +1,20 @@
 <template>
   <div class="pexpr" :class="classObj" id="id">
-    <div v-if="labeled" class="self">
+    <div v-if="labeled" class="self" :class="selfClassObj">
       <trace-label :traceNode="traceNode" :minWidth="minWidth"
                    @hover="onHover" @unhover="onUnhover" @click="onClick"
                    @showContextMenu="onShowContextMenu" />
+      <semantics-result v-if="hasResult" :traceNode="traceNode"
+                        @styleUpdate="updateSelfStyle"/>
     </div>
     <div v-if="!isLeaf" ref="children"
          class="children" :class="{vbox: vbox}"
          :hidden="collapsed">
       <trace-element v-for="child in childrenToRender"
                      :id="child.id" :traceNode="child.traceNode" :context="child.context"
                      :currentLR="child.currentLR" :measureInputText="measureInputText"
-                     :isInVBox="child.isInVBox" :eventHandlers="eventHandlers">
+                     :isInVBox="child.isInVBox" :eventHandlers="eventHandlers"
+                     :showResult="hasResult">
```

Comment:
I'm a bit confused here why we need both `showResult` and `hasResult`. Do they mean something different?

Also, do we need an extra guard here (e.g. `ohmEditor.options.enableSemanticsEditor`) to make sure that the semantics editor is not enabled in the visualizer-only mode?